qemu/tests/virtio_fs_host_owner_win.py

```diff
@@ -0,0 +1,196 @@
+import os
+import re
+import time
+import random
+import string
+
+import aexpect
+
+from avocado.utils import process
+
+from virttest import env_process
+from virttest import error_context
+from virttest import utils_misc
+from virttest import utils_test
+from virttest.qemu_devices import qdevices
+
+from provider import virtio_fs_utils
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test virtio-fs by sharing the data between host and guest.
+    Steps:
+        1. Create shared directories on the host.
+        2. Create a common user on host.
+        3. Start virtiofs daemon with root or the common user.
+        4. Change the shared directory's owner to a common user 'test' and give
+        all permission to other users.
+        5. Boot a guest with virtiofs options.
+        6. Log into guest and start virtiofs service.
+        7. Edit the registry to set the owner value and restart virtiofs service.
+        8. Create a file on the shared dir in guest.
+        9. Check the file from host side to check the file's UID/GID.
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+
+    def get_user_uid_gid(user_name):
+        """
+        Get user's UID and GID on host.
+        """
+        output = process.system_output("id %s" % user_name,
+                                       shell=True).decode().strip()
+        pattern = r"uid=(\d*).*gid=(\d*)"
+        match_string = re.findall(pattern, output)
+        uid_user = match_string[0][0]
+        gid_user = match_string[0][1]
+        return uid_user + ":" + gid_user
+
+    def enable_uid_gid(uid_gid_value):
+        """
+        Enable UID:GID argument to virtiofs service.
+        """
+        enable_cmd = params["viofs_owner_enable_cmd"] % uid_gid_value
+        s, o = session.cmd_status_output(enable_cmd)
+        if s:
+            test.fail("Fail command: %s. Output: %s." %
+                      (enable_cmd, o))
+        error_context.context("Restart virtiofs service after modify"
+                              " the registry.", test.log.info)
+        virtio_fs_utils.stop_viofs_service(test, params, session)
+        virtio_fs_utils.start_viofs_service(test, params, session)
+        time.sleep(1)
+
+    def check_file_uid_gid(volume_letter, expect_id):
+        """
+        Check file UID and GID on host.
+        """
+        error_context.context("Create a file in shared dir.", test.log.info)
+        file_name = "file_" + ''.join(random.sample(string.ascii_letters +
+                                                    string.digits, 3))
+        guest_file = volume_letter + ":\\" + file_name
+        session.cmd("dir z:\\")
```

Comment:
```suggestion
```
Is this line necessary? My suggestion: remove this line.
If you prefer to keep this line here, dir `Z:` should be gained dynamically because the shared dir may be changed.

Reply:
Deleted it, it's debug cmd.